pkg/v1/tkg/web/src/app/views/landing/vsphere-wizard/resource-step/resource-step.component.ts

```diff
@@ -116,23 +125,16 @@ export class ResourceStepComponent extends StepFormDirective implements OnInit {
                     }
                     if (source === TkgEventType.GET_VM_FOLDERS) {
                         this.resurrectField('vmFolder',
-                            [Validators.required, this.validationService.isValidNameInList(data.map(vmFolder => vmFolder.name))]);
+                            [Validators.required, this.validationService.isValidNameInList(
+                                data.map(vmFolder => vmFolder.name))], data.length === 1 ? data[0].name : '');
```

Comment:
nit: this format is a bit dense to parse (for me). I would tend to do the `data.map(vmFolder => vmFolder.name))]` first, and then not even add the validator in the case where `data.length === 1`